tasks/agent.py

```diff
@@ -754,6 +757,84 @@ def should_retry_bundle_install(res):
     return False
 
 
+def _compute_build_metrics(ctx, overall_duration):
+    # We only want to generate those metrics from the CI
+    src_dir = os.environ.get('CI_PROJECT_DIR')
+    if not src_dir:
+        return
+    series = []
+    timestamp = int(datetime.now().timestamp())
+    with open(f'{src_dir}/omnibus/pkg/build-summary.json') as summary_json:
+        j = json.load(summary_json)
+        for software, metrics in j['build'].items():
+            series.append(
+                {
+                    'metric': 'datadog.agent.build.duration',
+                    'points': [{'timestamp': timestamp, 'value': metrics['build_duration']}],
+                    'tags': [
+                        f'software:{software}',
+                        f'cached:{metrics["cached"]}',
+                        f'job:{os.environ.get("CI_JOB_NAME_SLUG")}',
+                    ],
+                    'unit': 'seconds',
+                    'type': 0,
+                }
+            )
+        series.append(
+            {
+                'metric': 'datadog.agent.build.total',
+                'points': [{'timestamp': timestamp, 'value': overall_duration}],
+                'tags': [f'job:{os.environ.get("CI_JOB_NAME_SLUG")}'],
+                'unit': 'seconds',
+                'type': 0,
+            }
+        )
+        if "strip" in j:
+            series.append(
+                {
+                    'metric': 'datadog.agent.build.strip',
+                    'points': [{'timestamp': timestamp, 'value': j['strip']}],
+                    'tags': [f'job:{os.environ.get("CI_JOB_NAME_SLUG")}'],
+                    'unit': 'seconds',
+                    'type': 0,
+                }
+            )
+
+        for packager, duration in j['packaging'].items():
+            series.append(
+                {
+                    'metric': 'datadog.agent.package.duration',
+                    'points': [{'timestamp': timestamp, 'value': duration}],
+                    'tags': [
+                        f'job:{os.environ.get("CI_JOB_NAME_SLUG")}',
+                        f'packager:{packager}',
+                    ],
+                    'unit': 'seconds',
+                    'type': 0,
+                }
+            )
+    dd_api_key = ctx.run(
+        'aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.datadog_api_key_org2 --with-decryption --query "Parameter.Value" --out text',
+        hide=True,
+    ).stdout.strip()
+    cmd = ' '.join(
+        (
+            'curl -X POST "https://api.datadoghq.com/api/v2/series"',
+            '-H "Accept: application/json"',
+            '-H "Content-Type: application/json"',
+            f'-H "DD-API-KEY: {dd_api_key}"',
+            '-d @-',
+        )
+    )
+    try:
+        series_json = json.dumps({'series': series})
+        ctx.run(cmd, hide='both', echo=False, in_stream=StringIO(series_json))
```

Comment:
I'm wondering what is the rule, could we prefer using `requests` rather than an curl wrapped in `ctx.run` (e.g. is it less secure somehow)

Reply:
I'd even consider using Boto3 rather than shelling out as well

Reply:
I don't know if there's a clear rule, but using requests makes much more sense here IMO.

Thanks for the suggestion!